Lib/Search/EntityRepository.php

```diff
@@ -176,7 +176,15 @@ public function delete($entity, $refresh = false)
      */
     public function deleteByQuery($query)
     {
-        $this->_sm->getClient()->removeAll($this->_class, $query);
+        $queryCriteria = $query->getQuery();
+        $results = $this->scanBy($queryCriteria);
+        foreach ($results as $collection) {
+            foreach ($collection as $entity) {
+                $this->delete($entity, false);
+            }
+            $this->_sm->flush();
+        }
+        $this->_sm->clear();
```

Comment:
That makes sense to me. Removing the clear() call has no effect on any of my test results. Will remove it from here.